packages/rollup-core/test/app/l2-provider-compatibility.spec.ts

```diff
@@ -0,0 +1,193 @@
+import {
+  add0x,
+  buildJsonRpcError,
+  ExpressHttpServer,
+  getLogger,
+  isObject,
+  JsonRpcRequest,
+  JsonRpcResponse,
+  logError,
+  Logger,
+} from '@eth-optimism/core-utils'
+import { JsonRpcProvider } from 'ethers/providers'
+import { keccak256 } from 'ethers/utils'
+import { monkeyPatchL2Provider } from '../../src/app'
+
+const nodeLogger: Logger = getLogger('mock-node', true)
+
+const port = 9991
+const hostname = '0.0.0.0'
+const senderAddress = '0xa7d9ddbe1f17865597fbd27ec712455208b6b76d'
+
+class MockNode extends ExpressHttpServer {
+  constructor(_hostname: string = hostname, _port: number = port) {
+    super(_port, _hostname)
+  }
+
+  protected initRoutes(): void {
+    this.app.post('/', async (req, res) => {
+      return res.json(await this.handleRequest(req))
+    })
+  }
+
+  /**
+   * Handles the provided request, returning the appropriate response object
+   * @param req The request to handle
+   * @returns The JSON-stringifiable response object.
+   */
+  protected async handleRequest(
+    req: any
+  ): Promise<JsonRpcResponse | JsonRpcResponse[]> {
+    let request: JsonRpcRequest
+
+    try {
+      request = req.body
+
+      const txResponse = {
+        blockHash:
+          '0x1d59ff54b1eb26b013ce3cb5fc9dab3705b415a67127a003c3e61eb445bb8df2',
+        blockNumber: '0x5daf3b', // 6139707
+        from: '0xa7d9ddbe1f17865597fbd27ec712455208b6b76d',
+        gas: '0xc350', // 50000
+        gasPrice: '0x4a817c800', // 20000000000
+        hash:
+          '0x88df016429689c079f3b2f6ad39fa052532c56795b733da78a91ebe6a713944b',
+        input: '0x68656c6c6f21',
+        nonce: '0x15', // 21
+        to: '0xf02c1c8e6114b1dbe8937a39260b5b0a374432bb',
+        transactionIndex: '0x41', // 65
+        value: '0xf3dbb76162000', // 4290000000000000
+        v: '0x25', // 37
+        r: '0x1b5e176d927f8e9ab405058b2d2457392da3e20f328b16ddabcebc33eaac5fea',
+        s: '0x4ba69724e8f69de52f0125ad8b3c5c2cef33019bac3249e2c0a2192766d1721c',
+        l1MessageSender: senderAddress,
```

Comment:
I guess eventually we'll need to also include the other fields like `queueOrigin` , etc. Or encode it in the calldata.

Oh actually something to note (that's not immediately related to this) is that we might want to have some kind of flag in L2Geth which either 1) excludes all these extra fields, or 2) encodes them in the calldata. The reason is because when running things like The Graph or block explorers, they might fail here for the same reason and therefore would require modified providers.

I guess a 3rd option is to modify them to actually parse this info, but that seems like a bit more work off the bat

Reply:
Sounds good. Definitely open to changing it. Am I reading this correctly to say that there is no action currently required, though?